clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -134,6 +134,16 @@ public void readInto(ByteBuffer buffer, int position) throws IOException {
      * @return A sliced wrapper on this message set limited based on the given position and size
      */
     public FileRecords slice(int position, int size) throws IOException {
+        SliceRange range = sliceRange(position, size);
+        return new FileRecords(file, channel, range.start, range.end, true);
+    }
+
+    public UnalignedFileRecords sliceUnaligned(int position, int size) {
```

Comment:
Can we add method documentation? I think the key thing to note is that this method is reserved for cases where the position is not necessarily on an offset boundary. It is worth mentioning the `FetchSnapshot` use case explicitly.